gprofiler/profilers/dotnet.py

```diff
@@ -0,0 +1,157 @@
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+import functools
+import os
+import signal
+import subprocess
+from pathlib import Path
+
+from threading import Event
+from typing import Any, Dict, List, Optional
+from granulate_utils.linux.process import process_exe
+from psutil import Process
+from gprofiler.exceptions import ProcessStoppedException, StopEventSetException
+from gprofiler.gprofiler_types import ProfileData
+from gprofiler.metadata.application_metadata import ApplicationMetadata
+from gprofiler.log import get_logger_adapter
+from gprofiler.profilers.profiler_base import ProcessProfilerBase
+from gprofiler.profilers.registry import register_profiler
+from gprofiler.utils import pgrep_maps, random_prefix, removed_path, resource_path, run_process
+from gprofiler.utils.speedscope import load_speedscope_as_collapsed
+logger = get_logger_adapter(__name__)
+
+
+class DotnetMetadata(ApplicationMetadata):
+
+    _DOTNET_VERSION_TIMEOUT = 3
+
+    @functools.lru_cache(4096)
+    def _get_dotnet_version(self, process: Process) -> str:
+        if not os.path.basename(process_exe(process)).startswith("dotnet"):
+            # TODO: for dynamic executables, find the ruby binary that works with the loaded libruby, and
+            # check it instead. For static executables embedding libruby - :shrug:
+            raise NotImplementedError
+        version = self.get_exe_version(process)  # not using cached version here since this wrapper is a cache
+        return version
+
+    def make_application_metadata(self, process: Process) -> Dict[str, Any]:
+        # ruby version
+        version = self._get_dotnet_version(process)
+
+        metadata = {"dotnet_version": version}
+
+        metadata.update(super().make_application_metadata(process))
+        return metadata
+
+
+@register_profiler(
+    "dotnet",
+    possible_modes=["dotnet-trace", "disabled"],
+    supported_archs=["x86_64", "aarch64"],
```

Comment:
Doesn't support `aarch64` currently. No need to support it in MVP. Remove from this list, and please update the `Architecture support` list in readme - add dotnet and mark as x86 :heavy_check_mark: and aarch64 :x: